src/composer.ts

```diff
@@ -384,24 +384,22 @@ export class Composer<C extends Context> implements MiddlewareObj<C> {
             if (cmd.startsWith("/")) {
                 throw new Error(
                     `Do not include '/' when registering command handlers (use '${
-                        cmd.substr(1)
+                        cmd.substring(1)
                     }' not '${cmd}')`,
                 );
             }
             const set = cmd.indexOf("@") === -1 ? noAtCommands : atCommands;
             set.add(cmd);
         });
-        return this.on(":entities:bot_command").filter(
+        return this.on("message:entities:bot_command").filter(
             (ctx): ctx is CommandContext<C> => {
-                const msg = ctx.message ?? ctx.channelPost;
-                const txt = msg.text ?? msg.caption;
-                const entities = msg.entities ?? msg.caption_entities;
```

Comment:
This not only removes support for commands in channels but also for commands on media messages with captions?

Not sure if that even works normally, never tried that.

Reply:
grammY used to support commands in caption in its early days, but a lot of people in the community complained that it is not intuitive for command handlers to trigger when e.g. an image caption contains a command. Hence, the support was remove by changing the filter query from `::bot_command` to `:entities:bot_command`.

The code that was removed here was essentially dead code.

Reply:
You can mix it, it's perfect!

Reply:
Mix? Do you mean merge?